pkg/provider/provider.go

```diff
@@ -49,6 +54,27 @@ type Legacy struct {
 func NewProviderCreator(jwtSecret string, legacy Legacy) ProviderCreator {
 	return func(user, password string) (Provider, error) {
 		switch {
+		case viper.IsSet("authorization-lambda"):
+			ok := CheckAuthorization(user, password, viper.GetString("authorization-lambda"), viper.GetString("aws-s3-region"))
+			if ok {
+				if legacy.JWT != nil {
```

Comment:
if this evaluates to false then no meaningful error will be provided

Reply:
Won't it drop through to the error returned on line 94? I can duplicate that line if you'd like (though I try to avoid that). Happy to do that whichever way you think is clearer.

Reply:
So I see two issues here:
* if `authorization-lambda` is set then this is the only authentication mechanism which is accepted. While I support this in general it should be more clear in the config which mechanism is actually to be used.
* `legacy.JWT == nil` if no AWS credentials are provided via the config file. For your implementation this is a configuration error which should cause an error on start up and not during runtime i.e. when a user actually tries to log in.

Reply:
Got it. I'll rework the logic in main.go. I think it also makes sense to disallow setting the "jwt-secret" as well, right?

Here's what I'm thinking.

If `authorization-lambda` is set, we require `aws-access-key-id`, `aws-secret-access-key` and `aws-s3-region`. A fatal error should be thrown if andy of `user`, `password` or `jwt-secret` are set.

Otherwise, your existing logic remains in place.

Reply:
Actually, I think I'll leave jwt-secret out of this since it sounds like it is still a bit experimental. It doesn't seem to be listed in the configuration options in README.md.

I'll restrict this to just "user" and "password".